drivers/spi/spi_sam.c

```diff
@@ -467,24 +687,35 @@ static const struct spi_driver_api spi_sam_driver_api = {
 	.release = spi_sam_release,
 };
 
-#define SPI_SAM_DEFINE_CONFIG(n)					\
-	static const struct spi_sam_config spi_sam_config_##n = {	\
-		.regs = (Spi *)DT_INST_REG_ADDR(n),			\
-		.periph_id = DT_INST_PROP(n, peripheral_id),		\
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),		\
-	}
-
-#define SPI_SAM_DEVICE_INIT(n)						\
-	PINCTRL_DT_INST_DEFINE(n);					\
-	SPI_SAM_DEFINE_CONFIG(n);					\
-	static struct spi_sam_data spi_sam_dev_data_##n = {		\
-		SPI_CONTEXT_INIT_LOCK(spi_sam_dev_data_##n, ctx),	\
-		SPI_CONTEXT_INIT_SYNC(spi_sam_dev_data_##n, ctx),	\
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(n), ctx)	\
-	};								\
-	DEVICE_DT_INST_DEFINE(n, &spi_sam_init, NULL,			\
-			    &spi_sam_dev_data_##n,			\
-			    &spi_sam_config_##n, POST_KERNEL,		\
+#define SPI_DMA_INIT(n)										\
+	.dma_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, tx)),				\
+	.dma_tx_channel = DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),				\
+	.dma_tx_perid = DT_INST_DMAS_CELL_BY_NAME(n, tx, perid),				\
+	.dma_rx_channel = DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),				\
+	.dma_rx_perid = DT_INST_DMAS_CELL_BY_NAME(n, rx, perid),
+
+#define SPI_SAM_USE_DMA(inst) DT_INST_DMAS_HAS_NAME(n, tx) && IS_ENABLED(CONFIG_SPI_SAM_DMA)
```

Comment:
Apparently this will always result in `0` because `inst != n`

Reply:
I tried to take fix it in my PR but [CI](https://github.com/zephyrproject-rtos/zephyr/actions/runs/3968156131) doesn't seem to like it..